src/parsers/manifest/dash/parse_adaptation_sets.ts

```diff
@@ -334,19 +344,18 @@ export default function parseAdaptationSets(
         isDub = true;
       }
 
-      const isClosedCaption = type === "text" &&
-                              accessibility != null &&
-                              isHardOfHearing(accessibility) ? true :
-                                                               undefined;
-      const isAudioDescription = type === "audio" &&
-                                 accessibility != null &&
-                                 isVisuallyImpaired(accessibility) ? true :
-                                                                     undefined;
-
-      const isSignInterpreted = type === "video" &&
-                                accessibility != null &&
-                                hasSignLanguageInterpretation(accessibility) ? true :
-                                                                               undefined;
+      const isClosedCaptionPredicate = (accessibility: IScheme) =>
+        type === "text" && isHardOfHearing(accessibility) ? true : undefined;
```

Comment:
As this is a predicate only used in a `some`, I don't think there's any advantage to returning `undefined` in that function, `some` will return `false` regardless.

But this also means that `isClosedCaption` will be set to `false` even for video/audio Adaptations.
It's not really a problem, it's even right in a sense so I have nothing against it. It also changes nothing for the API.
So your call here.

Also consequently we could remove that `type === "text"` from the predicate as this is a loop invariant, but again it's not like it's going to change anything much.
Again you choose :)

Reply:
Good point! I see what you're meaning and by removing the 'unnecessary' stuff the separate predicate functions is not longer needed, so I think that is a good alternative here! Just pushed a commit with those changes.